airbyte-integrations/connectors/destination-iceberg-v2/src/main/kotlin/io/airbyte/integrations/destination/iceberg/v2/IcebergTypesComparator.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.iceberg.v2
+
+import org.apache.iceberg.Schema
+import org.apache.iceberg.types.Type
+import org.apache.iceberg.types.Types
+
+/**
+ * Compares two Iceberg [Schema] definitions to identify:
+ * - New columns that do not exist in the "existing" schema.
+ * - Columns whose data types have changed.
+ * - Columns that no longer exist in the incoming schema (i.e., removed).
+ * - Columns that changed from required to optional.
+ */
+class IcebergTypesComparator {
+
+    /**
+     * Represents the differences between two Iceberg schemas.
+     *
+     * @property newColumns list of column names that are new in the incoming schema.
+     * @property updatedDataTypes list of column names whose types differ between incoming and
+     * existing schema.
+     * @property removedColumns list of column names that no longer exist in the incoming schema.
+     * @property newlyOptionalColumns list of column names that changed from required (existing) to
+     * optional (incoming).
+     */
+    data class ColumnDiff(
```

Comment:
Would it make sense here to have a `hasChanges()` method on this for ease of use?

Then we can change:

```
assertThat(diff.newColumns).isEmpty()
assertThat(diff.updatedDataTypes).isEmpty()
assertThat(diff.removedColumns).isEmpty()
assertThat(diff.newlyOptionalColumns).isEmpty()
```

Reply:
done